src/ring_theory/power_basis.lean

```diff
@@ -339,10 +285,58 @@ end
 
 /-- The power basis `1, root f, ..., root f ^ (d - 1)` for `adjoin_root f`,
 where `f` is an irreducible polynomial over a field of degree `d`. -/
-noncomputable def power_basis {f : polynomial K} (f_irr : irreducible f) :
+noncomputable def power_basis (hf : f ≠ 0) :
   power_basis K (adjoin_root f) :=
 { gen := root f,
   dim := f.nat_degree,
-  is_basis := power_basis_is_basis f_irr }
+  is_basis := power_basis_is_basis hf }
+
+lemma finite_dimensional (hf : f ≠ 0) : finite_dimensional K (adjoin_root f) :=
+finite_dimensional.of_fintype_basis (power_basis hf).is_basis
```

Comment:
Could you split this into a `power_basis.finite_dimensional` part and an `adjoin_root.finite_dimensional`? Then we could also add a `lemma power_basis.findim : findim R S = pb.dim`. (If this suggestion gives too many complications, let me know and I will fix it later in my own PR.)

Reply:
Done (although it looks like the `power_basis.finite_dimensional` was already there).